src/Arcus.EventGrid/Extensions/CloudEventExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using GuardNet;
+using Newtonsoft.Json.Linq;
+
+// ReSharper disable once CheckNamespace
+namespace CloudNative.CloudEvents
+{
+    /// <summary>
+    /// Add extensions on the <see cref="CloudEvent"/> SDK model.
+    /// </summary>
+    public static class CloudEventExtensions
+    {
+        /// <summary>
+        /// Gets the typed data payload from the <see cref="CloudEvent.Data"/> by parsing to a specified <typeparamref name="TData"/> type.
+        /// </summary>
+        /// <typeparam name="TData">The resulting event data type.</typeparam>
+        /// <param name="cloudEvent">The event from which the event data should be parsed.</param>
+        public static TData GetPayload<TData>(this CloudEvent cloudEvent)
+        {
+            Guard.NotNull(cloudEvent, nameof(cloudEvent));
+
+            if (cloudEvent.Data is null)
```

Comment:
Parsing of `.Data` is same as `EventGridEventExtensions.GetPayload`, maybe we can restructure it?

Reply:
It's a different class both in assemblies outside our control; but just by change has the same property called `.Data`. It's some duplication. We could use some kind of utility class for this; and pass allong the `object`.

Reply:
I'm fine with either, was just wondering - I'll leave it up to you to judge!

Reply:
I'll leave it like this. Seems a bit overkill. If it changes in some way or we have to do something else; we can reconsider it.

Reply:
Ok!